docs/designs/api-design.md

```diff
@@ -0,0 +1,343 @@
+
+Refer to the [Data Model Doc](./data-models.md) for source of truth details about each entity
+referrred in the doc here.
+
+## NOTE
+
+- All the APIs use the following prefix: `api.example.com/v1/`. The prefix is skipped in the following descriptions for simplicity.
+
+- Authentication parameters is not shown the following API definitions for simplicity
+
+- Common HTTP Status like `401 Unauthorized` or `500 _internal Server Error` are not shown here for simplicity
+
+- We try to use standard REST HTTP verbs as much as possible. Any deviations are described clearly in the API.
+
+- For simplicity, we are creating APIs around the business logic and not the underlying data models.
+  This is a deviation from best-practices. But it keeps the frontend logic fairly simple. In the backend
+  implementation, we will have high-level business logic methods that map to lower level data-model specific methods
+  to interface with the database
+
+- As much as possiible, we have tried to separate an object's data from the way it is visually
+  presented on the screen. This should simplify the backend logic and hopefully keep the
+  data storage de-coupled from the visual presentation. This should also allow the UI to be
+  rendered quickly without waiting for the entire data model to be parsed & processed.
+
+- All dates returned by the API are in UTC. UI is responsible for converting to local
+  timezone
+
+- All keys in the API request/response use Camel Case starting with a lower case letter.
+
+- In response JSONs, the core data elements are enclosed in a top-level key
+  representing the entity returned by the response. Example: In the `Get Programs API`, all
+  contents of the Program entity are stored within a top-level key called "program" -
+  `{"program": {...}}`. This is a more extensible design allowing us to send other metadata
+  along with the core entity.
+
+- All APIs have an `_internal` blob containing metadata used for backend's house keeping purposes.
+
+## User
+
+Get details about the logged in user
+
+`GET /user`
+
+**Response**
+
+```json
+{
+  "user": {
+    "id": "<uuid>",
+    "firstName": "<string>",
+    "lastName": "<string>",
+    "email": "<string>",
+    "organization": {
+      "id": "<uuid>",
+      "name": "<string>"
+    },
+    "workspaces": [
+      {
+        "id": "<uuid>",
+        "name": "<string>"
+      }
+    ]
+  },
+  "ui": {
+    // Shows the ID of the workspace that the user logs in by default.
+    "defaultWorkspaceId": "<uuid>"
+  }
+}
+```
+
+## Logout
+
+`POST /logout`
+
+**Request**
+None
+
+**Response**
+
+None
+
+## Get Workspace
+
+Get all details necessary to display the workspace page.
+
+`GET /workspace/<workspaceId>`
+
+**Request**
+
+None
+
+**Response**
+
+```json
+{
+  "workspace": {
+    "id": "<uuid>",
+    "name": "<string>",
+    "programs": [
+      {
+        "id": "<uuid>",
+        "name": "<string>",
+        "description": "<string>"
+      }
+    ],
+    "disease": {
```

Comment:
diseases are linked to workspaces or program? I think it should be linked to program.